File: frontend/lib/api/access.ts
import { apiFetch } from "@/lib/api/client";
import { grantDiagnosisAccess } from "@/lib/utils/accessGate";

interface AccessVerifyResponse {
  token: string;
  token_type: "bearer";
  expires_at: string;
}

export async function verifyDiagnosisCode(code: string): Promise<AccessVerifyResponse> {
  const response = await apiFetch<AccessVerifyResponse>("/api/access/verify", {
    method: "POST",
    auth: false,
    body: JSON.stringify({ code }),
  });

  grantDiagnosisAccess(response.token);
  return response;
}

File: backend/app/schemas/access.py
"""Schemas for diagnosis access verification."""
from __future__ import annotations

from pydantic import BaseModel, Field


class AccessVerifyRequest(BaseModel):
    code: str = Field(min_length=1)


class AccessVerifyResponse(BaseModel):
    token: str
    token_type: str = "bearer"
    expires_at: str
